rtbkit/plugins/augmentor_interface/http_augmentor_interface.cc

```diff
@@ -0,0 +1,486 @@
+            const PotentialBidder & bidder = group[j];
+            const AgentConfig & config = *bidder.config;
+            for (unsigned k = 0;  k < config.augmentations.size();  ++k) {
+                const std::string & name = config.augmentations[k].name;
+                augmentors.insert(name);
+                entry->augmentorAgents[name].insert(bidder.agent);
+            }
+        }
+    }
+
+    // Find which ones are actually available...
+    GcLock::SharedGuard guard(allAugmentorsGc);
+    const AllAugmentorInfo * ai = allAugmentors;
+    
+    //XXX : fix nemi
+    ExcAssert(ai);
+
+    for(auto it = augmentors.begin(); it != augmentors.end(); ++it){
+            // Augmentor we need to run
+            //cerr << "augmenting with " << it2->name << endl;
+            recordEvent("augmentation.request");
+            std::string eventName = "augmentor." + *it + ".request";
+            recordEvent(eventName.c_str());
+            
+            entry->outstanding.insert(*it);
+    }
+
+    if(entry->outstanding.empty()){
+        // No augmentors required... run the auction straight away
+        onFinished(info);
+    }else{
+        inbox.push(entry);
+    }
+}
+
+void HttpAugmentorInterface::registerLoopMonitor(LoopMonitor *monitor) const {
+    monitor->addMessageLoop("httpAugmentorInterfaceLoop", &loop);
+}
+
+void HttpAugmentorInterface::doAugmentation(const std::shared_ptr<Entry> & entry){
+    Date now = Date::now();
+
+    if (augmenting.count(entry->info->auction->id)) {
+        std::stringstream ss;
+        ss << "AugmentationLoop: duplicate auction id detected "
+            << entry->info->auction->id << std::endl;
+        std::cerr << ss.str();
+        recordHit("duplicateAuction");
+        std::cerr << "duplicate auction" << std::endl;
+        return;
+    }
+
+    bool sentToAugmentor = false;
+
+
+    for (auto it = entry->outstanding.begin(), end = entry->outstanding.end();
+         it != end;  ++it)
+    {
+        auto & aug = *augmentors[*it];
+        const AugmentorInstanceInfo* instance = pickInstance(aug);
+        if (!instance) {
+            recordHit("augmentor.%s.skippedTooManyInFlight", *it);
+            std::cerr << "skippedTooManyInFlight" << std::endl;
+            continue;
+        }
+        recordHit("augmentor.%s.instances.%s.request", *it, instance->path);
+
+        std::set<std::string> agents = entry->augmentorAgents[*it];
+        for(auto it = agents.begin(); it != agents.end(); ++it)
+            entry->info->auction->request->ext["agents"].append(*it);
+
+        
+        std::string name = *it;
+        std::string aid = entry->info->auction->id.toString();
+        Date date = Date::now();
+        std::string path = instance->path;
+
+        HttpRequest::Content reqContent {
+                entry->info->auction->request->toJsonStr(), "application/json" };
+
+        RestParams headers { { "X-Openrtb-Version", "2.1" } };
```

Comment:
For the record, we'll need to fix it and put the real version (solution for that should come along the upstream that will soon be merged)